src/main/java/org/logstash/filters/parser/CasualISO8601Parser.java

```diff
@@ -87,6 +87,17 @@ public Instant parse(BigDecimal value) {
 
   @Override
   public Instant parseWithTimeZone(String value, String timezone) {
-    return parse(value);
+    RuntimeException lastException = null;
+    for (DateTimeFormatter parser : parsers) {
+      try {
+        DateTimeZone tz = DateTimeZone.forID(timezone);
```

Comment:
No - will move.
The line will throw a IllegalArgumentException is the interpolated TZ string is not found but we don't do anything about that in the JodaParser class. The event will get tagged with failure but they will not know exactly why. I will create an issue for this.